### DISCUSSION TITLE
cannot open swagger /docs in protected route

### DISCUSSION THREAD

**Discussion by chawza** [fastapi/fastapi#3356] [Questions] [answered]
I added a middleware to my new sub-app that check whether all incoming request has valid token or not

```
app = FastAPI()
@app.middleware('http')
async def check_jwt_token(request: Request, next):
    try:
        authorization = request.headers['Authorization']
        token = authorization.split(' ')[1]
        Auth.validate(token)
        respond = await next(request)
        return respond
    except Exception:
        return JSONResponse(content={'message': 'invalid auth token'}, status_code=status.HTTP_401_UNAUTHORIZED)

//
all endpoints
//
```
but when I go to the /docs route, it won't open because I don't include the token in the header

how do I exclude the /docs route from the middleware in this particular app
the app will be mounted into the main app in main.py

or, if there is a better or proper way to do it, I want hear it

**Comment by r-bar** [comment]
I don't think you can exclude routes from middleware. Middleware gets triggered before the request even gets routed! The good news is it looks like the logic you are using can be captured as a dependency instead of middleware. Take a look at [this section of the docs](https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/#handle-jwt-tokens). FastAPI even provides special pre-built dependencies for doing this type of auth logic.

To keep things simple here is a direct translation of the code you posted into dependency form. This should still allow your docs to be exposed without checking the auth token.

```py
from fastapi import FastAPI, Depends, Header

async def check_jwt_token(authorization: str = Header(...)):
    try:
        token = authorization.split(' ')[1]
        Auth.validate(token)
    except Exception:
        return JSONResponse(content={'message': 'invalid auth token'}, status_code=status.HTTP_401_UNAUTHORIZED)

app = FastAPI(dependencies=[Depends(check_jwt_token)])

@app.get('/')
async def my_authorized_route():
    return

# all other endpoints
```

**Comment by YuriiMotov** [comment, marked as answer]
Better use approach with dependencies how it was advised above.
But if you want to keep using middleware, you can filter paths using `request.scope['path']`.
